src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockExpiredException;
+use Symfony\Component\Lock\Exception\LockStorageException;
+use Symfony\Component\Lock\Exception\NotSupportedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * MongoDbStore is a StoreInterface implementation using MongoDB as store engine.
+ *
+ * @author Joe Bennett <joe@assimtech.com>
+ */
+class MongoDbStore implements StoreInterface
+{
+    private $mongo;
+    private $options;
+    private $collection;
+
+    /**
+     * @param \MongoDB\Client $mongo
+     * @param array           $options
+     *
+     * database:    The name of the database [required]
+     * collection:  The name of the collection [default: lock]
+     * initialTtl:  The expiration delay of locks in seconds [default: 300.0]
+     *
+     * CAUTION: This store relies on all client and server nodes to have
+     * synchronized clocks for lock expiry to occur at the correct time.
+     * To ensure locks don't expire prematurely; the ttl's should be set with enough
+     * extra time to account for any clock drift between nodes.
+     *
+     * A TTL index MUST BE used on MongoDB 2.2+ to automatically clean up expired locks.
+     *
+     *     db.lock.ensureIndex(
```

Comment:
I suggest to provide a `public function createCollection` like in https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Cache/Traits/PdoTrait.php#L82 or https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Session/Storage/Handler/PdoSessionHandler.php#L209 or https://github.com/symfony/symfony/pull/27456/files

Reply:
I've added a method to `createTTLIndex()`. There's no need in MongoDB to create the collection, they're created on the fly by the database.